### ISSUE TITLE
Warning "Failed to load module err"

### ISSUE THREAD

**Issue by Joerg-L** [infracost/infracost#1730]
We observe in the logs of infracost fowllowing message

`level=warning parser=terraform_hcl msg="Failed to load module err: missing module with source 'registry.terraform.io/git::git@ssh.dev.azure.com:v3/<<AZURE_DEVOPS_ORG>>/<<AZURE_DEVOPS_PROJECT>>/<<AZURE_DEVOPS_REPO>> git::git@ssh.dev.azure.com:v3/<<AZURE_DEVOPS_ORG>>/<<AZURE_DEVOPS_PROJECT>>/<<AZURE_DEVOPS_REPO>>' -  try to 'terraform init' first"`

We have already run `terraform init` in the step before with follwoing outcome

```
Initializing modules...
Downloading git::ssh://git@ssh.dev.azure.com/v3/<<AZURE_DEVOPS_ORG>>/<<AZURE_DEVOPS_PROJECT>>/<<AZURE_DEVOPS_REPO>> for spoke...
- spoke in .terraform/modules/spoke
Downloading git::ssh://git@ssh.dev.azure.com/v3/<<AZURE_DEVOPS_ORG>>/<<AZURE_DEVOPS_PROJECT>>/<<AZURE_DEVOPS_REPO>> for spoke.peering_spoke_with_hub...
- spoke.peering_spoke_with_hub in .terraform/modules/spoke.peering_spoke_with_hub
Downloading git::ssh://git@ssh.dev.azure.com/v3/<<AZURE_DEVOPS_ORG>>/<<AZURE_DEVOPS_PROJECT>>/<<AZURE_DEVOPS_REPO>> for spoke.snet...
- spoke.snet in .terraform/modules/spoke.snet
Downloading git::ssh://git@ssh.dev.azure.com/v3/<<AZURE_DEVOPS_ORG>>/<<AZURE_DEVOPS_PROJECT>>/<<AZURE_DEVOPS_REPO>> for spoke.vnet...
- spoke.vnet in .terraform/modules/spoke.vnet

Initializing the backend...
```

so the message in the log seems to be wrong.

We are running infracost with following command

```
- script: infracost breakdown --path . --show-skipped --out-file=/tmp/infracost.json --format json   
  displayName: Run Infracost Output to file
  workingDirectory: '$(configroot)'
  env:
    INFRACOST_CURRENCY: EUR
```

**Comment by aliscott** [comment]
Thanks for reporting @Joerg-L 🙏. Do you have the HCL code of the module block that's causing this?

**Comment by aliscott** [comment]
As a workaround in the meantime if it helps, you can generate a plan JSON and use that as an input to Infracost, something like this:
```yml
    steps:
      - task: TerraformInstaller@0  # This can be obtained by installing the Microsoft Terraform extension: https://marketplace.visualstudio.com/items?itemName=ms-devlabs.custom-terraform-tasks
        displayName: Install Terraform

      # Generate the Terraform plan JSON
      - bash: |
          terraform init
          terraform plan -out plan.save
          terraform show -json plan.save > plan.json
        displayName: Generate plan JSON
        workingDirectory: $(TF_ROOT)

      - task: InfracostSetup@1
        displayName: Setup Infracost
        inputs:
          apiKey: $(infracostApiKey)

      # Generate an Infracost diff and save it to a JSON file.
      - bash: |
          infracost diff --path=$(TF_ROOT)/plan.json \
                         --format=json \
                         --out-file=/tmp/infracost.json
        displayName: Generate Infracost diff

      # Posts a comment to the PR using the 'update' behavior.
      # This creates a single comment and updates it. The "quietest" option.
      # The other valid behaviors are:
      #   delete-and-new - Delete previous comments and create a new one.
      #   hide-and-new - Minimize previous comments and create a new one.
      #   new - Create a new cost estimate comment on every push.
      # See https://www.infracost.io/docs/features/cli_commands/#comment-on-pull-requests for other options.
      - bash: |
          infracost comment github --path=/tmp/infracost.json \
                                   --github-token=$(githubToken) \
                                   --pull-request=$(System.PullRequest.PullRequestNumber) \
                                   --repo=$(Build.Repository.Name) \
                                   --behavior=update
        displayName: Post Infracost Comment
 ```
 
 More [examples are here](https://github.com/infracost/infracost-azure-devops/tree/master/examples/plan-json)

**Comment by Joerg-L** [comment]
> Do you have the HCL code of the module block that's causing this?

Here you are...

```
module "spoke" {
  source = "git::git@ssh.dev.azure.com:v3/<<DEVOPS_ORG>>/<<DEVOPS_PROJECT>>/<<DEVOPS_REPO>>"

  spoke_name = "spoke-example"
  location = "West Europe"

  address_space = ["xxx.xxx.xxx/24"]

  subnets = {
    "snet-one"  = ["xxx.xxx.xxx/26"]
    "snet-two"  = ["xxx.xxx.xxx/26"]
    "snet-three" = ["xxx.xxx.xxx/26"]
  }

  hub_vnet_name = local.hub_vnet_name
  hub_vnet_id = local.hub_vnet_id
  hub_vnet_resource_group = local.hub_vnet_resource_group_name 
}
```

**Comment by aliscott** [comment]
Thank you, we just had another user reproduce this issue as well!

**Comment by aliscott** [comment]
@Joerg-L when we're debugging this it might be useful to know how this module shows in your `.terraform/modules/modules.json` file, if you have that.

**Comment by Joerg-L** [comment]
here you are

`{"Modules":[{"Key":"","Source":"","Dir":"."},{"Key":"spoke","Source":"git::ssh://git@ssh.dev.azure.com/v3/<<DEV_OPS_ORG>>/<<DEVOPS_PROJECT>>/<<DEVOPS_REPO>>","Dir":".terraform/modules/spoke"},{"Key":"spoke.peering_spoke_with_hub","Source":"git::ssh://git@ssh.dev.azure.com/v3/<<DEV_OPS_ORG>>/<<DEVOPS_PROJECT>>/<<DEVOPS_REPO>>","Dir":".terraform/modules/spoke.peering_spoke_with_hub"},{"Key":"spoke.snet","Source":"git::ssh://git@ssh.dev.azure.com/v3/<<DEV_OPS_ORG>>/<<DEVOPS_PROJECT>>/<<DEVOPS_REPO>>","Dir":".terraform/modules/spoke.snet"},{"Key":"spoke.vnet","Source":"git::ssh://git@ssh.dev.azure.com/v3/<<DEV_OPS_ORG>>/<<DEVOPS_PROJECT>>/<<DEVOPS_REPO>>","Dir":".terraform/modules/spoke.vnet"}]}`

**Comment by aliscott** [comment]
Thanks @Joerg-L I've found a potential issue and put a fix in #1731. Another workaround until that is released is to change the format of your source attribute from:

```hcl
    source = "git::git@ssh.dev.azure.com:v3/<<DEVOPS_ORG>>/<<DEVOPS_PROJECT>>/<<DEVOPS_REPO>>"
```

to:
```hcl
    source = "git::ssh://git@ssh.dev.azure.com/v3/<<DEVOPS_ORG>>/<<DEVOPS_PROJECT>>/<<DEVOPS_REPO>>"
```

**Comment by hugorut** [comment]
Hey @Joerg-L [the v0.10.2](https://github.com/infracost/infracost/releases/tag/v0.10.2) build has been released and it contains @aliscott's fix for this issue. Let us know if you have any further problems. Thanks
